### PR TITLE
docs: list supported and unsupported Git features

### PR DESCRIPTION
**Pull request by martinvonz**

### CODE REVIEW

File: docs/git-compatibility.md [hunk 1]
```diff
@@ -9,6 +9,37 @@ See `jj help git` for help about the `jj git` family of commands, and e.g.
...
+
+* Configuration: No, the only configuration from Git (e.g. in `~/.gitconfig`)
+  that's respected is the configuration of remotes. Feel free to file a bug
+  if you miss any particular configuration options.
+* Authentication: Partial. Only `ssh-agent` or a password-less key file at
+  `~/.ssh/id_rsa` (and only at exactly that path).
+* Branches: Yes. You can read more about
+  [how branches work in Jujutsu](branches.md)
+  and [how they interoperate with Git](#branches).
+* Tags: Partial. You can check out tagged commits by name, but you cannot
```
Comment (arxanas) [inline docs/git-compatibility.md:22]: Tag objects or lightweight tags? For lightweight tags, if supported, can you modify/delete them?

**Review by arxanas — COMMENTED** [review]
Other topics you might want to mention, in some order:

* Merge commits (and maybe octopus merges?).
* Rebasing. (Maybe also commit deduplication as a part of rebasing? If a commit is merged into the main branch upstream, can you rebase on top on of it and skip the appropriate commit?)
* Orphan branches/unborn HEAD.
* Detached HEAD.
* Whether there's a reflog-like strucure. (Maybe the operation log is a superior replacement?)
* The staging area/partial commits. Obviously one of the features is that there's no more staging area, so you might want to explain how to achieve equivalent workflows for partial commits or moving changes across branches.
* Whether jj can be used with a bare Git repository backing it.
* Garbage collection/what kind of objects are cleaned up, if any.
* Object packing (both whether you can save disk space and whether jj will incur periodic repacking pauses, or perhaps whether jj generates a lot of object churn).

File: docs/git-compatibility.md (hunk 1 above)
Comment (martinvonz) [reply docs/git-compatibility.md]: > Tag objects or lightweight tags? For lightweight tags, if supported, can you modify/delete them?

Both. Updated the text.

File: docs/git-compatibility.md [hunk 2]
```diff
@@ -9,6 +9,53 @@ See `jj help git` for help about the `jj git` family of commands, and e.g.
...
+  native implementation, so please report a bug if you notice any difference
+  compared to `git`.  
+* .gitattributes: No. There's [#53](https://github.com/martinvonz/jj/issues/53)
+  about adding support for at least the `eol` attribute.
+* Merge commits: Yes, including octopus merges (i.e. with more than 2 parents).
+* Detached HEAD: Yes. Jujutsu supports anonymous branches, so this is a natural
+  state.
+* Orphan branch: Yes. Jujutsu has a virtual root commit that appears as parent
+  of all commits Git would call "root commits".
+* Staging area. Kind of. The staging area will be ignored. For example,
```
Comment (arxanas) [inline docs/git-compatibility.md:40]:
```suggestion
* Staging area: Kind of. The staging area will be ignored. For example,
```

Worth adding a link to https://github.com/martinvonz/jj/blob/main/docs/git-comparison.md#the-index in this section.

Comment (martinvonz) [reply docs/git-compatibility.md]: Good point, will add the link.

**Comment by spoerri** [comment]
Might want to mention cherry picking

**Comment by martinvonz** [comment]
> Might want to mention cherry picking

For my understanding, what do you think I should say about it?

I ask because I think it's more relevant in https://github.com/martinvonz/jj/blob/main/docs/git-comparison.md (which I need to expand quite a bit). There I could explain how to achieve cherry-picking with `jj`. It could also make sense to mention in this doc how `jj` behaves after you do a `git cherry-pick`. So I think it could be documented in both places.

**Comment by spoerri** [comment]
> more relevant in https://github.com/martinvonz/jj/blob/main/docs/git-comparison.md

OK, I guess this page could be more about which git _repo_ features are supported.

> how to achieve cherry-picking with jj

That'd be great, thank you

**Comment by martinvonz** [comment]
> > more relevant in https://github.com/martinvonz/jj/blob/main/docs/git-comparison.md
> 
> OK, I guess this page could be more about which git _repo_ features are supported.

Exactly, that was my intent anyway. I'll try to fill in `git-comparison.md` soon. Perhaps it turns out to be too unclear what the difference between the two docs is and I'll have to merge them or restructure, but for now I'll continue writing them as two separate docs.

> > how to achieve cherry-picking with jj
> 
> That'd be great, thank you

I've added that to this PR.

**Comment by martinvonz** [comment]
I've updated `git-comparison.md` as well now. Please take a look if you have time.

**Review by arxanas — APPROVED** [review]
Looks good on my end. You might want to bold the subject for each bullet point to make it easier to skim through.

**Comment by martinvonz** [comment]
> Looks good on my end.

Thanks for reviewing.

>  You might want to bold the subject for each bullet point to make it easier to skim through.

That seems like a good idea. I've updated both `git-compatibility.md` and `git-comparison.md`.

**Pull request merged**